src/controllers/bookController.ts

```diff
@@ -1,85 +1,139 @@
+      }
+      const newBook = await book.save();
+      const author = await Author.findById(newBook.author);
+      if (author) {
+        author.books.push(newBook.id);
+        await author.save();
+      }
+      res.redirect("book");
+      console.log(newBook);
+    } catch (error) {
+      console.log(error);
+    }
+  };
 
-    public static saveBook = async (req:Request, res:Response) => {
-        console.log(req.body);
-        console.log(req.files);
-        const book = new Book({
-            title: req.body.title, 
-            author: req.body.author,
-            publishDate: req.body.publishDate,
-            pageCount: req.body.pageCount,
-            description: req.body.description,
-        });
-        if(req.files){
-            book.cover = Buffer.from((req.files as any).cover.data, 'base64');
-            book.coverType = (req.files as any).cover.mimetype;
+  public static updateBook = async (req: Request, res: Response) => {
+    try {
+      const book = await Book.findById(req.params.id);
+      if (book) {
+        book.title = req.body.title;
+        book.author = req.body.author;
+        book.publishDate = req.body.publishDate;
+        book.pageCount = req.body.pageCount;
+        book.description = req.body.description;
+        if (req.files) {
+          book.cover = Buffer.from((req.files as any).cover.data, "base64");
+          book.coverType = (req.files as any).cover.mimetype;
         }
-        const newBook = await book.save();
-        res.redirect('book');
-        console.log(newBook);
-    } 
-    
-    public static updateBook = async (req:any, res:Response) => {
-        const book:any = await Book.findById(req.params.id);
-        if(req.body){
-            book.title = req.body.title;
-            book.author = req.body.author;
-            book.publishDate = req.body.publishDate;
-            book.pageCount = req.body.pageCount;
-            book.description = req.body.description;
-            if(req.files){
-                book.cover = Buffer.from((req.files as any).cover.data, 'base64');
-                book.coverType = (req.files as any).cover.mimetype;
-            }
+        const savedBook = await book.save();
+        const author = await Author.findById(savedBook.author);
+        if (author && book.author != savedBook.author) {
+          author.books.push(savedBook.id);
+          await author.save();
         }
-        await book.save();
         this.renderBook(req, res);
-    } 
+      }
+    } catch (error) {
+      console.log(error);
+    }
+  };
 
-    public static editBook = async (req:any, res:Response) => {
-        const user = req.user;
-        const authors = await Author.find({});
-        const book:any = await Book.findById(req.params.id);
-        res.render('book/edit', {book:book, authors:authors, user:user});
+  public static editBook = async (req: Request, res: Response) => {
+    const user = req.user;
```

Comment:
I've fixed it using a custom property user in Request interface. but this is not reflected in the repository, since the edited index.d.ts file is located in the folder node_modules. I'm not sure if this is the best approach, but it allows me to easily get the user from the request without changing the Request type to any.

//C:\Users\Alexa\OneDrive\Desktop\solvd\javascript\library\node_modules\@types\express\index.d.ts
import {User} from "../models/user";

declare function e(): core.Express;

declare namespace e {

    interface Request {
        user?: User;
      }
      //old code
}

I've found it here:
https://stackoverflow.com/questions/37377731/extend-express-request-object-using-typescript?page=1&tab=scoredesc#tab-top
https://github.com/TypeStrong/ts-node/issues/715#issuecomment-526757308

Reply:
It’s not a good practice to make changes in the node_modules folder, this folder is usually not pushed into the remote repo.
This is because you’ll be the only one having that change and the ones pulling your code will not.
Please try finding a workaround.